tests/units/RedisMock.php

```diff
@@ -129,6 +129,37 @@ public function testSetGetDelExists()
             ->integer($redisMock->setnx("test-setnx-expire", "lala"))
                 ->isEqualTo(1);
 
+        //set with nx
+        $this->assert
+            ->string($redisMock->set('test-set-nx', 'value', ['nx']))
+            ->isEqualTo('OK');
+        $this->assert
+            ->integer($redisMock->set('test-set-nx', 'value', ['nx']))
+            ->isEqualTo(0);
+
+        //set with xx
+        $this->assert
+            ->integer($redisMock->set('test-set-xx', 'value', ['xx']))
+            ->isEqualTo(0);
+        $this->assert
+            ->string($redisMock->set('test-set-xx', 'value'))
+            ->isEqualTo('OK');
+        $this->assert
+            ->string($redisMock->set('test-set-xx', 'value2', ['xx']))
+            ->isEqualTo('OK');
+
+        //set with nx ex
+        $this->assert
+            ->string($redisMock->set('test-set-nx-ex', 'value', ['nx', 'ex' => 1]))
+            ->isEqualTo('OK');
+        $this->assert
+            ->integer($redisMock->set('test-set-nx-ex', 'value', ['nx', 'ex' => 1]))
+            ->isEqualTo(0);
+        sleep(2);
```

Comment:
Is it for debugging purpose? 😄 
Can you remove it?

Reply:
The test is specifically waiting for an expiration, so you need to sleep for 2 seconds to allow the expiration of 1 second to pass.

Reply:
OH… my bad, sorry 😅 👍